hooks/environment

```diff
@@ -12,6 +12,22 @@ check_command() {
   fi
 }
 
+# the buildkite agent redactor has been added from version 3.66.0 and on. So we need to check it.
+# https://github.com/buildkite/agent/releases/tag/v3.66.0
+check_buildkite_version() {
+  local version_output
+  version_output=$(buildkite-agent --version)
```

Comment:
This should be captured without breaking the spaces inside it:
```suggestion
  version_output="$(buildkite-agent --version)"
```

Reply:
I tried it and it does not break the spaces inside it. Also even if it did we need the semantic version and this is what we get from the regex, so it would make no difference, this is how the output looks like:
```
buildkite-agent version 3.96.0+10034.c43a98aeda00817f5f150d172b32f475071adf5c
```

Reply:
Via https://stackoverflow.com/a/10067297 I feel we should wrap it in quotes to "be safe"

Reply:
Changed in 2cab9997094d5f493b5fd50e2ac765f66a5e37de